### PR TITLE
Add cluster-api-provider-azure milestone v0.4.15

### PR DESCRIPTION
**Pull request by nader-ziada**
add capz v0.4.15 milestone

/cc @CecileRobertMichon

### CODE REVIEW

File: config/prow/plugins.yaml [hunk 1]
```diff
@@ -372,7 +372,7 @@ milestone_applier:
     release-0.3: v0.3
   kubernetes-sigs/cluster-api-provider-azure:
     master: v0.5.0
-    release-0.4: v0.4.14
+    release-0.4: v0.4.15
```
Comment (CecileRobertMichon) [inline config/prow/plugins.yaml:375]: thanks! this was an oversight from my part

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm
/approve

File: config/prow/plugins.yaml (hunk 1 above)
Comment (nader-ziada) [reply config/prow/plugins.yaml:375]: no worries!

**Pull request merged**